extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,1094 @@
+
+		// FakeUser
+		if ( $this->mFakeUserId != 0 ) {
+			$uss = new UserStatsService( $this->mFakeUserId );
+			$uss->calculateEditCountWiki();
+		} else {
+			// use OldUsername if FakeUser isn't set
+			$oldUser = User::newFromName( $this->mOldUsername );
+			$uss = new UserStatsService( $oldUser->getId() );
+			$uss->calculateEditCountWiki();
+		}
+	}
+
+	/**
+	 * @author Federico "Lox" Lucignano <federico@wikia-inc.com>
+	 * Performs action for cleaning up temporary data at the very end of a process
+	 */
+	public function cleanup() {
+		if ( $this->mFakeUserId ) {
+			$this->addLog( "Cleaning up process data in user option renameData for ID {$this->mFakeUserId}" );
+
+			$fakeUser = User::newFromId( $this->mFakeUserId );
+			$fakeUser->setGlobalAttribute( 'renameData', self::RENAME_TAG . '=' . $this->mNewUsername );
+			$fakeUser->saveSettings();
+			$fakeUser->saveToCache();
+		}
+
+		// TODO: Add a hook
+		$hookName = 'UserRename::Cleanup';
+		$this->addLog( "Broadcasting hook: {$hookName}" );
+		Hooks::run( $hookName, array( $this->mRequestorId, $this->mRequestorName, $this->mUserId, $this->mOldUsername, $this->mNewUsername ) );
+
+		$tasks = [];
+		if ( isset( $this->mLogTask ) ) {
+			$tasks[] = $this->mLogTask->getID();
+		}
+
+		$this->addMainLog( "finish", RenameUserLogFormatter::finish( $this->mRequestorName, $this->mOldUsername, $this->mNewUsername, $this->mReason, $tasks ) );
+	}
+
+	/**
+	 * Sends the internal log message to the specified destination
+	 *
+	 * @param $text string Log message
+	 * @param $arg1 mixed Multiple format parameters
+	 */
+	public function addLog( $text, $arg1 = null ) {
+		if ( func_num_args() > 1 ) {
+			$args = func_get_args();
+			$args = array_slice( $args, 1 );
+			$text = vsprintf( $text, $args );
+		}
+		foreach ( $this->mLogDestinations as $destinationEntry ) {
+			$logDestination = $destinationEntry[0];
+
+			/** @var BatchTask $logTask */
+			$logTask = $destinationEntry[1];
+			switch ( $logDestination ) {
+				case self::LOG_BATCH_TASK:
+					$logTask->log( $text );
+					break;
+				case self::LOG_OUTPUT:
+					echo $text . "\n";
+					break;
+//				case self::LOG_STANDARD:
+				default:
+					wfDebugLog( __CLASS__, $text );
+			}
+		}
+	}
+
+	/**
+	 * Logs the message to main user-visible log
+	 *
+	 * @param string $action
+	 * @param $text string Log message
+	 */
+	public function addMainLog( $action, $text ) {
+		StaffLogger::log(
+			'renameuser',
+			$action,
+			$this->mRequestorId,
+			$this->mRequestorName,
+			$this->mUserId,
+			$this->mNewUsername,
+			$text
+		);
+	}
+
+	/**
+	 * Adds log message in Special:Log for the current wiki
+	 *
+	 * @param $text string Log message
+	 */
+	public function addLocalLog( $text ) {
+		$log = new LogPage( 'renameuser' );
+		$log->addEntry( 'renameuser', Title::newFromText( $this->mOldUsername, NS_USER ), $text, array(), User::newFromId( $this->mRequestorId ) );
+	}
+
+	public function setRequestorUser() {
+		global $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app